Add jobs for PHP `8.3` to the CI

ste93cry opened the pull request:
With PHP `8.3` coming in November 2023, it's time to run the CI for that version as well so that we're ready for its release.

Jean85 commented on .github/workflows/static-analysis.yaml:
```diff
@@ -19,7 +19,7 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
-          php-version: '8.1'
+          php-version: '8.3'
```
This is failing, and PHP-CS-Fixer is slow to support newer language versions. We should stick to 8.2 here
```suggestion
          php-version: '8.2'
```

ste93cry replied:
>PHP-CS-Fixer is slow to support newer language versions

It makes me sad, but I agree. I wanted to run the tol on PHP `8.3` just for the sake of consistency with the other jobs, but I remember that in the past we had to wait on average a long time for a compatible version to be released

ste93cry commented:
@cleptric I think it's pointless to retest all PHP versions just to make sure they work without optional packages installed, so I removed some jobs from the GitHub Workflow and changed the matrix so that only the oldest and newest PHP versions of each major are tested. However, as you can see from the following image, some jobs are still flagged as required checks for the CI to pass. Maybe the repo settings need to be adjusted accordingly?

![immagine](https://github.com/getsentry/sentry-symfony/assets/1770485/71fddc17-3e5c-4ba4-b1e7-d88b3d6fc56d)

cleptric commented:
The required checks haven been removed.
